prov/efa/src/rdm/efa_rdm_ep_utils.c

```diff
@@ -231,18 +228,28 @@ int efa_rdm_ep_post_user_recv_buf(struct efa_rdm_ep *ep, struct efa_rdm_ope *rxe
 	pkt_entry->alloc_type = EFA_RDM_PKE_FROM_USER_BUFFER;
 	pkt_entry->flags = EFA_RDM_PKE_IN_USE;
 	pkt_entry->next = NULL;
+	pkt_entry->ep = ep;
 	/*
 	 * The actual receiving buffer size (pkt_size) is
 	 *    rxe->total_len - sizeof(struct efa_rdm_pke)
 	 * because the first part of user buffer was used to
 	 * construct pkt_entry. The actual receiving buffer
 	 * posted to device starts from pkt_entry->wiredata.
 	 */
-	pkt_entry->pkt_size = rxe->iov[0].iov_len - sizeof(struct efa_rdm_pke);
-
+	pkt_entry->pkt_size = rxe->total_len - sizeof *pkt_entry;
```

Comment:
During test I find `rxe->total_len` is not initialized before this function is called ....let's use `ofi_total_iov_len(rxe->iov, rxe->iov_count)`

Reply:
Shouldn't we initialize it instead?

Reply:
I remember it is expected to initialize that field during the progress engine. Because the total len aims to reflect the total length of data that efa received, it may be small than the recv buffer posted.

Reply:
Actually, I think we can just move the `FI_MSG_PREFIX` logic above this line. At that point, `rxe->total_len` would be set. WDYT?

Reply:
I moved the code block up. Let me know if you think it's insufficient.

Reply:
I think we just shouldn't update rxe->total_len here. It should be set as the received data size in the progress engine.